base/ltclass.dtx

```diff
@@ -1515,7 +1515,9 @@
 %<latexrelease>\IncludeInRelease{2021/06/01}%
 %<latexrelease>                 {\OptionNotUsed}{filter unused option list}%
 %<*2ekernel|latexrelease>
-\def\@remove@eq@value#1=#2\@nil{#1}
+\ExplSyntaxOn
+\def\@remove@eq@value#1=#2\@nil{\tl_trim_spaces:n{#1}}
```

Comment:
Should this be expanded at this stage?

Reply:
It's the same as the similar treatment in `ltkeys.dtx`, which prohibits further expansion on key names, see the `\exp_not:n` in `\@@_remove_equals:w`:
https://github.com/latex3/latex2e/blob/22375e7b5e90cc9638f62806f69a11aa532430e5/base/ltkeys.dtx#L496-L498
https://github.com/latex3/latex2e/blob/22375e7b5e90cc9638f62806f69a11aa532430e5/base/ltkeys.dtx#L388-L390

Also from a previous discussion https://github.com/latex3/latex2e/issues/580#issuecomment-855303853, the agreement seemed to be that there is only a one-step expansion on the whole raw option list.

Reply:
BTW the logic to remove optional `=value` part and trim spaces from remaining key name is currently implemented twice, once in `ltkeys.dtx` in pure expl3, and another one in `ltclass.dtx` in mixed 2e + expl3.

A follow-up code refactor (after this PR) can be done. Currently in building the LaTeX foramt, `ltclass.dtx` is loaded just before `ltkeys.dtx`.
https://github.com/latex3/latex2e/blob/22375e7b5e90cc9638f62806f69a11aa532430e5/base/format.ins#L210-L211